teku/src/main/java/tech/pegasys/teku/cli/subcommand/MigrateDatabaseCommand.java

```diff
@@ -0,0 +1,186 @@
+    }
+    // validate source database exists
+    final DatabaseVersion sourceDatabaseVersion =
+        confirmAndCheckOriginalDb(maybeOutputVersion.get());
+
+    final DatabaseMigrater dbMigrater =
+        DatabaseMigrater.builder()
+            .dataOptions(dataOptions)
+            .dataStorageOptions(dataStorageOptions)
+            .eth2NetworkOptions(eth2NetworkOptions)
+            .batchSize(batchSize)
+            .statusUpdater(SUB_COMMAND_LOG::display)
+            .build();
+
+    try {
+      dbMigrater.migrateDatabase(sourceDatabaseVersion, maybeOutputVersion.get());
+      SUB_COMMAND_LOG.display("SUCCESS.");
+      SUB_COMMAND_LOG.display(
+          "The original database is stored in: " + dbMigrater.getMovedOldBeaconFolderPath());
+      SUB_COMMAND_LOG.display("This can be removed once the new database has been verified.");
+    } catch (DatabaseMigraterError error) {
+      SUB_COMMAND_LOG.error("FAILED to migrate database: " + error.getMessage());
+      SUB_COMMAND_LOG.display(
+          "There is a partially created database at: " + dbMigrater.getNewBeaconFolderPath());
+      SUB_COMMAND_LOG.display("This is not in use and could be cleaned up.");
+      System.exit(1);
+    }
+  }
+
+  private DatabaseVersion confirmAndCheckOriginalDb(final DatabaseVersion databaseVersion) {
+    // validate source database exists
+    final DatabaseVersion sourceDatabaseVersion = validateSourceDatabase(databaseVersion);
+
+    displaySourceDatabaseDetails(sourceDatabaseVersion);
+    SUB_COMMAND_LOG.display("Requested database version: " + databaseVersion);
+    SUB_COMMAND_LOG.display("A beacon.new folder will be created with the new database");
+    SUB_COMMAND_LOG.display("If the data is moved successfully: ");
+    SUB_COMMAND_LOG.display(" - The existing beacon folder will become beacon.old");
+    SUB_COMMAND_LOG.display(" - The beacon.new folder will become beacon (the active database)");
+    SUB_COMMAND_LOG.display(
+        " - Once the new database is tested successfully, you can remove the old database");
+    SUB_COMMAND_LOG.display("This operation will need to happen while teku is not running.");
+    SUB_COMMAND_LOG.display("");
+    if (!confirmYes("Proceed with database migration (yes/no)?")) {
+      SUB_COMMAND_LOG.display("Operation cancelled.");
+      System.exit(0);
+    }
+    return sourceDatabaseVersion;
+  }
+
+  private boolean confirmYes(final String question) {
+    SUB_COMMAND_LOG.display(question);
+    Scanner scanner = new Scanner(System.in, Charset.defaultCharset().name());
+    final String confirmation = scanner.next();
+    return confirmation.equalsIgnoreCase("yes");
+  }
+
+  private void displaySourceDatabaseDetails(final DatabaseVersion sourceDatabaseVersion) {
+    final DataDirLayout dataDirLayout = DataDirLayout.createFrom(dataOptions.getDataConfig());
+
+    SUB_COMMAND_LOG.display(
+        "Current database path: " + dataDirLayout.getBeaconDataDirectory().resolve("db"));
```

Comment:
Not sure that we should be pointing to the `db` subdirectory here.
```suggestion
        "Current database path: " + dataDirLayout.getBeaconDataDirectory());
```